### PR TITLE
Fix treemap field detection (display-name aware) + remove deprecations

### PR DESCRIPTION
**Pull request by preira**
This PR fixes a common “Configure your query…” failure mode where the treemap panel can’t find the configured label/size/color fields after Grafana transformations (e.g., when fields are renamed via display name). It makes field selection more robust by matching against both raw field names and display-name variants, and keeps color optional unless explicitly configured—so the panel can render as long as label + size are present in a single frame.

In addition, it includes small refactors and cleanup to improve testability and reduce deprecated API usage.

Key changes

Robust field matching:
Match configured fields by [field.name](vscode-file://vscode-app/c:/Users/Jo%C3%A3o/AppData/Local/Programs/Microsoft%20VS%20Code/resources/app/out/vs/code/electron-browser/workbench/workbench.html) and display-name sources (e.g., [displayName](vscode-file://vscode-app/c:/Users/Jo%C3%A3o/AppData/Local/Programs/Microsoft%20VS%20Code/resources/app/out/vs/code/electron-browser/workbench/workbench.html), [displayNameFromDS](vscode-file://vscode-app/c:/Users/Jo%C3%A3o/AppData/Local/Programs/Microsoft%20VS%20Code/resources/app/out/vs/code/electron-browser/workbench/workbench.html), state display name).
Build “frame views” more deterministically so required fields are detected within a single data frame.
UX behavior:
Color field is optional unless explicitly configured; defaults to using size/standard coloring when not set.
Refactor for testability:
Extracted frame selection/build logic into a pure helper and added table-driven unit tests to cover matching behavior and edge cases.
Deprecation/lint cleanup:
Replace deprecated vector access patterns with index access.
Migrate away from deprecated [theme.v1](vscode-file://vscode-app/c:/Users/Jo%C3%A3o/AppData/Local/Programs/Microsoft%20VS%20Code/resources/app/out/vs/code/electron-browser/workbench/workbench.html) usage.
Replace deprecated [Select](vscode-file://vscode-app/c:/Users/Jo%C3%A3o/AppData/Local/Programs/Microsoft%20VS%20Code/resources/app/out/vs/code/electron-browser/workbench/workbench.html) with [Combobox](vscode-file://vscode-app/c:/Users/Jo%C3%A3o/AppData/Local/Programs/Microsoft%20VS%20Code/resources/app/out/vs/code/electron-browser/workbench/workbench.html) in the field selector editor.
How to test

Unit tests: yarn test:ci
Lint/typecheck: yarn lint and yarn typecheck
Manual (Grafana):
Add a query returning multiple fields and apply transforms (join/organize/rename).
Configure treemap label/size (and optionally color) and verify it renders even when the visible column names come from display-name metadata.
Notes

There may still be a tooling notice about TypeScript version compatibility with @typescript-eslint/* (doesn’t affect runtime behavior, but can be addressed separately if desired).

### CODE REVIEW

File: src/frameSelection.ts [hunk 1]
```diff
@@ -0,0 +1,70 @@
+import { Field, FieldType } from '@grafana/data';
```
Comment (fastfrwrd) [inline src/frameSelection.ts:1]:
```suggestion
import { DataFrame, Field, FieldMatcherID, FieldType, getFieldMatcher } from '@grafana/data';
```

File: src/frameSelection.ts [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+    field.name,
+    field.config?.displayName,
+    field.config?.displayNameFromDS,
+    (field.state as any)?.displayName,
+  ]
+    .map(normalize)
+    .filter(Boolean);
+
+  return candidates.includes(wantedNormalized);
+};
```
Comment (fastfrwrd) [inline src/frameSelection.ts:23]:
```suggestion
export const fieldMatches = (
  field: Field,
  wanted: string | undefined,
  frame: DataFrame,
  allFrames: DataFrame[]
): boolean => {
  const name = normalize(wanted);
  if (!name) {
    return false;
  }

  // byName matches the raw field name OR its resolved display name via
  // getFieldDisplayName (config.displayName -> displayNameFromDS -> labels/frame
  // disambiguation), which is exactly the resolution this helper was reproducing.
  return getFieldMatcher({ id: FieldMatcherID.byName, options: name })(field, frame, allFrames);
};
```

File: src/frameSelection.ts [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+  frame: { fields: Field[] },
+  type: FieldType,
+  wanted?: string
+): Field<T> | undefined => {
+  if (normalize(wanted)) {
+    return frame.fields.find((f) => f.type === type && fieldMatches(f, wanted)) as Field<T> | undefined;
+  }
+
+  return frame.fields.find((f) => f.type === type) as Field<T> | undefined;
+};
```
Comment (fastfrwrd) [inline src/frameSelection.ts:35]:
```suggestion
export const findFieldByTypeAndName = <T = unknown>(
  frame: DataFrame,
  allFrames: DataFrame[],
  type: FieldType,
  wanted?: string
): Field<T> | undefined => {
  if (normalize(wanted)) {
    return frame.fields.find((f) => f.type === type && fieldMatches(f, wanted, frame, allFrames)) as
      | Field<T>
      | undefined;
  }

  return frame.fields.find((f) => f.type === type) as Field<T> | undefined;
};
```

File: src/frameSelection.ts [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+      const color = colorByFieldExplicit
+        ? findFieldByTypeAndName<number>(frame, FieldType.number, options.colorByField)
+        : size ?? findFieldByTypeAndName<number>(frame, FieldType.number);
+
+      const groupBy = frame.fields.find((f) => fieldMatches(f, options.groupByField));
+
+      const labels =
+        options.labelFields
+          ?.map((name) => frame.fields.find((f) => fieldMatches(f, name)))
+          .filter((f): f is Field => Boolean(f)) ?? [];
```
Comment (fastfrwrd) [inline src/frameSelection.ts:57]:
```suggestion
export const buildFrameViews = (series: DataFrame[], options: TreemapOptions): FrameView[] => {
  return series
    .map((frame) => {
      const text = findFieldByTypeAndName<string>(frame, series, FieldType.string, options.textField);
      const size = findFieldByTypeAndName<number>(frame, series, FieldType.number, options.sizeField);

      // Color is optional unless explicitly configured.
      const colorByFieldExplicit = normalize(options.colorByField).length > 0;
      const color = colorByFieldExplicit
        ? findFieldByTypeAndName<number>(frame, series, FieldType.number, options.colorByField)
        : size ?? findFieldByTypeAndName<number>(frame, series, FieldType.number);

      const groupBy = frame.fields.find((f) => fieldMatches(f, options.groupByField, frame, series));

      const labels =
        options.labelFields
          ?.map((name) => frame.fields.find((f) => fieldMatches(f, name, frame, series)))
          .filter((f): f is Field => Boolean(f)) ?? [];
```

**Review by fastfrwrd — COMMENTED** [review]
Sorry it took so long to get you a review on this — thanks for your patience, and for the fix. The display-name matching is the right call.

One suggestion: `@grafana/data` already exports a helper that does this, so you don't have to maintain the matching yourself. `getFieldMatcher({ id: FieldMatcherID.byName, options: name })` returns a `(field, frame, allFrames) => boolean` that matches a field by its raw name or its display name. Internally it uses `getFieldDisplayName`, which covers the cases handled here — `config.displayName`, `displayNameFromDS`, label formatting, disambiguating names across frames — plus some Time/Value edge cases that are easy to miss.

One functional note: the current `fieldMatches` only sees a single frame, so it can't resolve the display names Grafana generates when multiple frames are present, which is probably part of what this PR is trying to fix. Passing `allFrames` into the matcher (the suggestions below thread it through) handles that.

Note that this shifts a few `frameSelection.test.ts` assertions — the matcher resolves each field to one canonical display name (config wins), so the per-variant checks and the 2-arg `fieldMatches(f, 'x')` calls will need updating once the signature becomes `(field, wanted, frame, allFrames)`.
